src/common/inc/fAppDownloadDialog.h

```diff
@@ -0,0 +1,98 @@
+///////////////////////////////////////////////////////////////////////////////////////
+// fAppDownloadDialog.h
+//
+// Copyright (c) 2018. All rights reserved.
+// Section of Biomedical Image Analysis
+// Center for Biomedical Image Computing and Analytics
+// Department of Radiology
+// Perelman School of Medicine
+// University of Pennsylvania
+//
+// Contact details: software@cbica.upenn.edu
+//
+// License Agreement: https://www.med.upenn.edu/sbia/software-agreement.html
+///////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef _fAppDownloadDialog_h_
+#define _fAppDownloadDialog_h_
+
+
+//#include "CAPTk.h"
+#include "ui_fAppDownloadDialog.h"
+#include <QDialog>
+#include <QNetworkAccessManager>
+#include <QNetworkRequest>
+#include <QNetworkReply>
+#include <QUrl>
+#include <QProgressDialog>
+#include <QFile>
+#include <QFileInfo>
+#include <QDir>
+#include <QMessageBox>
+
+/**
+\class fAppDownloadDialog
+
+\brief This class controls the elements in the DICOM converter
+*/
+class fAppDownloadDialog : public QDialog, private Ui::fAppDownloadDialog
+{
+  Q_OBJECT
+
+public:
+  fAppDownloadDialog();
+  ~fAppDownloadDialog();
+
+  // QString m_exe, m_dataDir, m_modelDir; // contains full path and exe name of dcm2nii
+  // std::string m_baseModelDir;
+
+  QString downloadPath;
+  QString extractPath;
+  QString fullPath;
+  QString appName;
+  std::string downloadLink;
+
+  QUrl url;
+  QNetworkAccessManager *manager;
+  QNetworkReply *reply;
+  QFile *file;

```

Comment:
Unless this _absolutely_ needs to be raw pointer, let's refrain from using it and allocate it on the stack.

Reply:
Got it